### ISSUE TITLE
[Bug]: AOI over 3 S1 SLC bursts not subsetting

### ISSUE THREAD

**Issue by Boorhin** [mobigroup/gmtsar#162]
Hi,
I have an AOI that spans 3 bursts and I was following the tutorial for the Golden Valley to test the code. 
![image](https://github.com/user-attachments/assets/6200811a-d560-457e-af32-7f6400dc5af2)

It seems that the AOI is ignored on the plots in some ways compared to the example
![image](https://github.com/user-attachments/assets/1d7c527a-fea6-4b8a-9d2e-f2ef92281839)
![image](https://github.com/user-attachments/assets/1b2b297b-85c1-462d-9344-0642b4b33a25)

However the PS function is calculated over the 3 bursts
![image](https://github.com/user-attachments/assets/d135bf64-6658-4364-af55-ee75fdec299f)

I am planning to cut that corner as it is not useful so I have only 2 bursts but just to let you know.

**System and software version:**
 - OS: Ubuntu 22.04
 - PyGMTSAR version: 2024.8.2.post3

**Comment by AlexeyPechnikov** [comment]
Yes, I see—the bounding box of the selected burst covers the area fully. You can add a buffer like AOI.buffer(0.05) to expand your Area of Interest (AOI). This issue is technically related to GMTSAR binaries used in Sentinel-1 preprocessing. I plan to develop my own Sentinel-1 preprocessor to address this and other issues (including saving significant disk space) when I have the time for this project.

**Comment by kpzt-sven** [comment]
While this solves the problem for this current example data the issue is not related to GMTSAR binaries. I think the issue lies in how `_reframe_subswath()` handles geometries and converts the azimuth indices.

Below are laid down visual geometries and analyses of the code for some other dataset.
The `_reframe_subswath()` utilizes many geometries:
- input argument (purple)
- minimum rotated rectangle (green)
- exterior (essentially identical to rotated one) which is used for getting the diagonals
- diag1 (cyan) and diag2 (thick neon red) to derive azimuth indices based on which the data is cut/extracted
- out['geometry'] the geometry returned by the `_reframe_subswath' (blue)

![Image](https://github.com/user-attachments/assets/3beb3ca0-67f9-4998-85d0-4b04543bf174)

Same as above but with the blue out geometry

![Image](https://github.com/user-attachments/assets/ba5cc532-f909-40ef-891a-25cb50c97e6f)

The SAR data not processed is because the `diag2` is selected as seen below. Although there is no difference between them, the IEEE754 format influenced difference would make the `diag2` shorter and selected.

![Image](https://github.com/user-attachments/assets/798df34a-70f9-4c7a-a6b7-ebc59e692949)

This would be alleviated by changing the if clause to:
```
            if diag1.length >= diag2.length:
                geometry = diag1
            else:
                geometry = diag2
```
And the results would be fine and filling the AOI completely

![Image](https://github.com/user-attachments/assets/7ab77402-4ac4-4bd8-91f2-b38aeb371a74)

Finally the correct solution should be one actually following the SAR geometry -orbit ascending or descending properties which would influence the diagonal selection and azimuth indices closer to scenes'/data SAR geometry.

**Comment by AlexeyPechnikov** [comment]
There are many cases when complicated geometries are not parsed correctly to azimuthal limits. You can just simplify your geometry or patch the function as you prefer.
